encoding.go

```diff
@@ -0,0 +1,89 @@
+package verkle
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+const ErrInvalidNodeEncoding = "invalid node encoding"
+
+func ParseNode(serialized []byte, tc *TreeConfig) (VerkleNode, error) {
+	elems, _, err := rlp.SplitList(serialized)
+	if err != nil {
+		return nil, err
+	}
+	c, err := rlp.CountValues(elems)
+	if err != nil {
+		return nil, err
+	}
+
+	if c == 1 {
```

Comment:
I advise a `swtich`/`case` and also to define some constants instead of just having numbers.